light_node/src/main.rs

```diff
@@ -159,16 +150,24 @@ fn block_on_actors(
 
     // initialize shell automaton manager
     let (mut shell_automaton_manager, rpc_shell_automaton_channel) = ShellAutomatonManager::new(
+        tezos_protocol_api.clone(),
```

Comment:
Here it is clonned, but then wrapped in Arc later, shouldn't you be wrapping it first and passing `Arc::clone(&wrapped)` instead?

Reply:
no I need mutable access to it from state machine.